net/wireshark3/Portfile

```diff
@@ -167,22 +168,37 @@ variant chmodbpf description {Enable Wireshark to access macOS capture devices}
     depends_run             port:wireshark-chmodbpf
 }
 
-variant python36 conflicts python37 python38 python39 description {Use python36 during build} {
-    depends_build-append    port:python36
-}
+set pythons_suffixes {36 37 38 39 310}
 
-variant python37 conflicts python36 python38 python39 description {Use python37 during build} {
-    depends_build-append    port:python37
+set pythons_ports {}
+foreach py_ver_nodot ${pythons_suffixes} {
+    lappend pythons_ports python${py_ver_nodot}
 }
 
-variant python38 conflicts python36 python37 python39 description {Use python38 during build} {
-    depends_build-append    port:python38
+foreach py_ver_nodot ${pythons_suffixes} {
+    set py_port python${py_ver_nodot}
+    set py_ver [string index ${py_ver_nodot} 0].[string range ${py_ver_nodot} 1 end]
+    set i [lsearch -exact ${pythons_ports} ${py_port}]
+    set py_conflicts [lreplace ${pythons_ports} ${i} ${i}]
+    variant ${py_port} description "Use python ${py_ver} during build" conflicts {*}${py_conflicts} "
+        depends_build-append      port:${py_port}
+    "
 }
 
-variant python39 conflicts python36 python37 python38 description {Use python39 during build} {
-    depends_build-append    port:python39
+# make sure one of the +python* variants is set
+set python_isset false
+foreach py_ver_nodot ${pythons_suffixes} {
+    if {!${python_isset}} {
+        set python_isset  [variant_isset python${py_ver_nodot}]
+    }
+}
```

Comment:
While it doesn't really matter much here as the "pythons_suffixes" is so short and I expect the common case is to not provide a python variant, I generally prefer to bail out of the loop as soon as we find that a python variant is set, like:
```foreach py_ver_nodot ${pythons_suffixes} {
    if { [variant_isset python${py_ver_nodot}] } {
        set python_isset true
        break
    }  
}
```
Thoughts?

Reply:
Sure the list is short, but `break` can't hurt, and it makes for cleaner code overall. I just edited in your suggestion :-)